app/Nova/RemoteAttendanceLink.php

```diff
@@ -0,0 +1,209 @@
+     *
+     * @var string
+     */
+    public static $title = 'id';
+
+    /**
+     * The logical group associated with the resource.
+     *
+     * @var string
+     */
+    public static $group = 'Meetings';
+
+    /**
+     * Indicates if the resource should be displayed in the sidebar.
+     *
+     * @var bool
+     */
+    public static $displayInNavigation = false;
+
+    /**
+     * The relationships that should be eager loaded on index queries.
+     *
+     * @var array<string>
+     */
+    public static $with = ['attendable'];
+
+    /**
+     * Indicates if the resource should be globally searchable.
+     *
+     * @var bool
+     */
+    public static $globallySearchable = false;
+
+    /**
+     * The list of recommended values for the note field.
+     *
+     * @var array<string>
+     *
+     * @phan-suppress PhanReadOnlyPublicProperty
+     */
+    public static $recommendedNotes = [
+        'Electrical Subteam', 'Mechanical/Mechatronics Subteam', 'Software Subteam', 'Whole Team', 'Training',
+        'Trainers Only', 'Discipline Core (n/a)', 'Event (n/a)',
+    ];
+
+    /**
+     * Get the fields displayed by the resource.
+     */
+    public function fields(Request $request): array
+    {
+        return [
+            MorphTo::make('Team/Event', 'attendable')
+                ->types([
+                    Event::class,
+                    Team::class,
+                ])
+                ->readonly(static function (Request $request): bool {
+                    return true;
+                }),
+
+            Text::make('Auto-redirecting Link', 'secret')
+                ->onlyOnDetail()
+                ->resolveUsing(static function (string $secret): string {
+                    return route('attendance.remote.redirect', ['secret' => $secret]);
+                })
+                ->canSee(static function (Request $request): bool {
+                    if (isset($request->resourceId)) {
+                        $resource = RAL::find($request->resourceId);
+                        if (null !== $resource && is_a($resource, RAL::class)) {
+                            return null !== $resource->redirect_url;
+                        }
+                    }
+
+                    return false;
+                })
+                ->readonly(static function (Request $request): bool {
+                    return true;
+                }),
+
+            Text::make('Non-redirecting Link', 'secret')
+                ->onlyOnDetail()
+                ->resolveUsing(static function (string $secret): string {
+                    return route('attendance.remote', ['secret' => $secret]);
+                })
+                ->readonly(static function (Request $request): bool {
+                    return true;
+                }),
+
+            Text::make('Secret')
+                ->onlyOnForms()
+                ->default(hash('sha256', random_bytes(64)))
```

Comment:
Same as earlier. Should we factor that out to a helper class somewhere?

Reply:
Eventually that would probably be good.